docs/intro/sdk-design.md

```diff
@@ -0,0 +1,95 @@
+# Cosmos SDK design overview
+
+The Cosmos SDK is a framework that facilitates the development of secure state-machines on top of Tendermint. At its core, the SDK is a boilerplate implementation of the ABCI in Golang. It comes with a `multistore` to persist data and a `router` to handle transactions. 
+
+Here is a simplified view of how transactions are handled by an application built on top of the Cosmos SDK when transferred from Tendermint via `DeliverTx`
+(the `CheckTx` process is the same without enforcing state changes):
+
+1. Decode transactions received from the Tendermint consensus engine (remember that Tendermint only deals with `[]bytes`). 
+2. Extract messages from transactions and do basic sanity checks.
+3. Route each message to the appropriate module so that it can be processed. 
+4. Commit the state changes.
+
+The application also enables you to generate transactions, encode them and pass them to the underlying Tendermint engine to broadcast them. 
+
+## `baseapp`
+
+`baseApp` is the boilerplate implementation of the ABCI of the Cosmos SDK. It comes with a `router` to route transactions to their respective module. The main `app.go` file of your application will define your custom `app` type that will embed `baseapp`. This way, your custom `app` type will automatically inherit all the ABCI methods of `baseapp`. See an example of this in the [SDK application tutorial](https://github.com/cosmos/sdk-application-tutorial/blob/master/app.go#L27).
+
+The goal of `baseapp` to provide a secure interface between the store and the extensible state machine while defining as little about that state machine as possible (staying true to the ABCI).
+
+For more on `baseapp`, please click [here](../concepts/baseapp.md).
+
+## Multistore
+
+The Cosmos SDK provides a multistore for persisting  state. The multistore allows developers to declare any number of [`KVStores`](https://github.com/blocklayerhq/chainkit). These `KVStores` only accept the `[]byte` type as value and therefore any custom structure needs to be marshalled using [go-amino](https://github.com/tendermint/go-amino) before being stored.
```

Comment:
```suggestion
 The Cosmos SDK provides a multistore for persisting state. The multistore allows developers to declare any number of [`KVStores`](https://github.com/blocklayerhq/chainkit). These `KVStores` only accept the `[]byte` type as value and therefore any custom structure needs to be marshalled using [go-amino](https://github.com/tendermint/go-amino) before being stored.
```

Reply:
Also I think this is the wrong link, did you mean to link to ChainKit?

Reply:
No, will update